### PR TITLE
CB-9839 Add gzip support to file-transfer.download on wp8

### PR DESCRIPTION
**Pull request by sviluppatoribk**
Actually WP platform does not support gzip decompression if you send "Accept-Encoding:gzip" header on download request (Android and iOS platform altready support it). 

With this fix we have a drastically improvement of download performance on WP8.x platform (only when server support compression of course).

I have included Microsoft System.IO.Compression.dll (https://www.nuget.org/packages/Microsoft.Bcl.Compression/3.9.85)
for compatibility with WP8.0 platform.

### CODE REVIEW

**Comment by sviluppatoribk** [comment]
Please can you tell me what error code you get instead of 401? Test 6 of file-transfer-plugin call "http://cordova-filetransfer.jitsu.com/download_basic_auth" url and on my run it return correct error code.. I have see there are many pull request failed for this test case, you are able to call that url from test machine?

**Comment by daserge** [comment]
@sviluppatoribk, thanks for your contribution!
I have filed a Jira issue for this:
https://issues.apache.org/jira/browse/CB-9839

In order this to be merged could you please rebase, squash and rename commit and PR to `"CB-9839 Add gzip support to file-transfer.download on wp8"`?
Also please sign the ICLA: http://www.apache.org/dev/new-committers-guide.html#cla

gzip-case also needs an update in terms of progress event as length becomes non computable; the code change can be the following:
```csharp
bool useGzip = encodingHeader.Contains("gzip");
using (Stream responseStreamGz = useGzip ? new GZipStream(response.GetResponseStream(), CompressionMode.Decompress) : response.GetResponseStream())
...
if (buffer.Length > 0 && !reqState.isCancelled)
{
    writer.Write(buffer);

    if (useGzip)
    {
        DispatchFileTransferProgress(0, 0, callbackId);
    }
    else
    {
        DispatchFileTransferProgress(bytesRead, totalBytes, callbackId);
    }
}
...
```

File: src/wp/FileTransfer.cs [hunk 1]
```diff
@@ -697,31 +698,39 @@ private void downloadCallback(IAsyncResult asynchronousResult)
                     {
                         long totalBytes = response.ContentLength;
                         int bytesRead = 0;
-                        using (BinaryReader reader = new BinaryReader(response.GetResponseStream()))
+
+
+                        // Ambrogelli 4 gzip!
```
Comment (daserge) [inline src/wp/FileTransfer.cs]: Please translate to English.

**Comment by daserge** [comment]
Ping, @sviluppatoribk

**Comment by sviluppatoribk** [comment]
Done

**Comment by maverickmishra** [comment]
With the new features introduced in [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), this plugin is not needed any more. Migrating from this plugin to using the new features of [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), is explained in this [Cordova blog post](https://cordova.apache.org/blog/2017/10/18/from-filetransfer-to-xhr2.html).

**Pull request closed without merging**